services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -170,7 +163,44 @@ export class DataAnalyticsService {
         query = Prisma.sql`SELECT ${selectClause}, COUNT(id) AS "transactionCount" FROM "TransactionRecord" tr WHERE ${whereClause} GROUP BY ${groupByClause}`;
     }
 
-    return await this._executeQuery(query);
+    return await this._executeQuery<Array<Record<string, unknown>>>(query);
+  }
+
+  async evaluateHighTransactionTypePercentage({
+    projectId,
+    transactionType,
+    subjectColumn,
+    minimumCount,
+    minimumPercentage,
+    timeAmount,
+    timeUnit,
+  }: HighTransactionTypePercentage) {
+    return await this._executeQuery<Array<{ counterpartyId: string }>>(Prisma.sql`
+      WITH "transactionsData" AS (
+        SELECT
+          "${Prisma.raw(subjectColumn)}",
+          COUNT(*) AS "transactionCount",
+          COUNT(*) FILTER (WHERE "transactionType" = '${Prisma.raw(
+            transactionType,
+          )}') AS "filteredTransactionCount"
+        FROM
+          "TransactionRecord"
+        WHERE
+          "projectId" = ${projectId}
+          AND "transactionDate" >= CURRENT_DATE - INTERVAL '${Prisma.raw(
+            `${timeAmount} ${timeUnit}`,
+          )}'
```

Comment:
and also here please use: `Prisma.sql`

Reply:
It doesn't work with `Prisma.sql`, we use `Prisma.raw` in the previous rules as well.